quic/s2n-quic/src/provider/event/console_perf.rs

```diff
@@ -0,0 +1,296 @@
+    event_loop_wakeup: AtomicU64,
+    timeout: AtomicU64,
+    pto_count: AtomicU64,
+    max_pacing_rate: AtomicU64,
+    max_delivery_rate: AtomicU64,
+}
+
+impl Counters {
+    pub fn print_header(&self) {
+        println!(
+            "Tx Rate\t\
+            Rx Rate\t\
+            Max Cwnd\t\
+            Max Inflight\t\
+            Lost Packets\t\
+            Wakeups\t\
+            Duration\t\
+            Max RTT\t\
+            Max SRTT\t\
+            PTO Count\t\
+            Max Pacing Rate\t\
+            Max Delivery Rate"
+        );
+    }
+
+    pub fn print(&self, duration: Duration) {
+        // The goodput of data transmitted to the peer
+        let send_progress = self.send_progress.swap(0, Ordering::Relaxed);
+        let send_rate = rate(send_progress, duration);
+        // The goodput of data received from the peer
+        let receive_progress = self.receive_progress.swap(0, Ordering::Relaxed);
+        let receive_rate = rate(receive_progress, duration);
+        // The maximum congestion window observed during the interval
+        let max_cwnd = self.max_cwnd.swap(0, Ordering::Relaxed);
+        let max_cwnd = bytes(max_cwnd);
+        // The maximum amount of unacknowledged data in flight during the interval
+        let max_bytes_in_flight = self.max_bytes_in_flight.swap(0, Ordering::Relaxed);
+        let max_bytes_in_flight = bytes(max_bytes_in_flight);
+        // The maximum round trip time observed during the interval
+        let max_rtt = self.max_rtt.swap(0, Ordering::Relaxed);
+        let max_rtt = Duration::from_nanos(max_rtt);
+        // The maximum smoothed (weighted average) round trip time observed during the interval
+        let max_smoothed_rtt = self.max_smoothed_rtt.swap(0, Ordering::Relaxed);
+        let max_smoothed_rtt = Duration::from_nanos(max_smoothed_rtt);
+        // The number of packets recorded lost during the interval
+        let lost_packets = self.lost_packets.swap(0, Ordering::Relaxed);
+        // The number of event loop wakeups recorded during the interval
+        let wakeups = self.event_loop_wakeup.swap(0, Ordering::Relaxed);
+        // The duration of the latest event loop wakeup
+        let duration = self.timeout.swap(0, Ordering::Relaxed);
+        let duration = Duration::from_nanos(duration);
+        // The number of packet time out events observed during the interval
+        let pto_count = self.pto_count.swap(0, Ordering::Relaxed);
+        // The maximum rate at which packets are paced out observed during the interval
+        let max_pacing_rate = self.max_pacing_rate.swap(0, Ordering::Relaxed);
+        let max_pacing_rate = rate(max_pacing_rate, Duration::from_secs(1));
+        // The maximum estimate of bandwidth observed during the interval. Only output for BBRv2
+        let max_delivery_rate = self.max_delivery_rate.swap(0, Ordering::Relaxed);
+        let max_delivery_rate = rate(max_delivery_rate, Duration::from_secs(1));
+        println!(
```

Comment:
One thing we should note is we probably can't break this once apps start using it (in case there's some automation around specific fields). We can always add new fields but the order and presence of the existing ones needs to be stable.

Reply:
I think we either need to document what we're committing to or put a note inline about us not committing to the format.

Reply:
Added a comment in a few places noting that the fields are subject to change